src/assets/tabs.js

```diff
@@ -1,52 +1,92 @@
+var tabClass = "react-tabs__tab";
+
 function attachTabs() {
-  var tabClass = "react-tabs__tab";
-  var lu = document.createElement("lu");
-  lu.classList.add(`${tabClass}-list`);
-
-  var sections = document.querySelectorAll("section.tabbed-narrative");
-  sections.forEach(function(section, idx) {
-    var li = `
-    <li class="${tabClass}"
-        data-tab="${section.id}"
-        role="tab"
-        aria-selected="false"
-        aria-disabled="false"
-        aria-controls="tabs-${idx + 1}"
-    >
-      ${section.title}
-    </li>`;
-    lu.innerHTML += li;
-  });
-  document.getElementById("tabs").appendChild(lu);
-  var lis = document.querySelectorAll(`li.${tabClass}`);
-  [].forEach.call(lis, function(li) {
-    li.addEventListener("click", function() {
-      lis.forEach(function(li_) {
-        li_.classList.remove(`${tabClass}--selected`);
-      });
-      li.classList.add(`${tabClass}--selected`);
-      sections.forEach(function(section_) {
-        section_.style.display = "none";
+  /* This function is run after the page loads. It iterates 
+   * through the "dehydrated" tab containers on the page and
+   * "hydrates" their tab link bar and contents sections, selecting
+   * the first tab by default.
+   * A click handler is registered for each tab link, which calls
+   * activateTab(). activateTab() is also run at the end of this
+   * function if there is a hash fragment in the URL, directing that
+   * the specified tab should be activated and brought into focus.
+   */
+
+  var containers = document.querySelectorAll("div.tabs-container");
+  containers.forEach(function(container) {
+    var ul = document.createElement("ul");
+    ul.classList.add(`${tabClass}-list`);
+    var sections = container.querySelectorAll("section");
+    sections.forEach(function(section, idx) {
+      var li = document.createElement("li");
+      li.classList.add(tabClass);
+      li.setAttribute("role", "tab");
+      li.setAttribute("aria-controls", section.id);
+      // Activate the first tab by default
+      if (idx == 0) {
+        section.style.display = "block";
+        li.classList.add(`${tabClass}--selected`);
+        li.setAttribute("aria-selected", "true");
+      } else {
+        section.style.display = "none";
+        li.setAttribute("aria-selected", "false");
+      }
+      li.appendChild(document.createTextNode(section.title));
+      li.addEventListener("click", function() {
+        activateTab(null, section.id);
       });
-      document.querySelector(
-        `section#${li.getAttribute("data-tab")}`
-      ).style.display =
-        "block";
+      ul.appendChild(li);
     });
+    container.querySelector("div.wrapper").children[0].appendChild(ul);
   });
-  // Show contents of first tab by default
-  lis[0].click();
+
+  if (window.location.hash) activateTab();
```

Comment:
This line would be better at the bottom before attaching the `onHashChange` listener, but it's not likely to matter in this case.

Reply:
I'm happy to move it there, though in that case I assume the order would be the following?
```
attachTabs();
if (window.location.hash) activateTab();
window.addEventListener("hashchange", activateTab, false);
```